src/queries.cpp

```diff
@@ -140,13 +145,16 @@ void perftest(const std::string &index_filename,
         mapper::map(wdata, md, mapper::map_flags::warmup);
     }
 
-    std::vector<float> thresholds;
+    std::vector<Threshold> thresholds(queries.size(), 0.0);
     if (thresholds_filename) {
         std::string t;
         std::ifstream tin(*thresholds_filename);
+    size_t idx = 0;
         while (std::getline(tin, t)) {
-            thresholds.push_back(std::stof(t));
+            thresholds[idx] = std::stof(t);
+        idx += 1;
         }
+        assert(idx == queries.size());
```

Comment:
There are some weird indents here, but more importantly, can we simply use `transform`?
```cpp
std::copy(std::istream_iterator<float>(tin), std::istream_iterator<float>(), thresholds.begin());
```